### ISSUE TITLE
isTransitioning holds value of true at end of transition

### ISSUE THREAD

**Issue by charles-salmon** [react-navigation/react-navigation#3963]
### Current Behaviour

- I am using `withNavigation` to wrap a `headerLeft` component. This component is to represent a "Menu" icon at the bottom of a stack, or a "Back" icon otherwise.
- When entering a stack at some page other than that specified in the `initialRouteName` and `pop()`'ing off the stack until the bottom of the stack is reached, `props.navigation.state.index === 0 && !props.navigation.state.isTransitioning` is evaluated to `false`. That is `isTransitioning` holds a value of `true` at the end of the transition.

### Expected Behaviour

- I would expect `isTransitioning` to hold a value of `false` at the end of the transition.

### How to Reproduce

1. [Open this Snack](https://snack.expo.io/B15lZL2sf).
2. Open drawer menu.
3. Click "Navigate to page three."
4. Click back icon.
5. Observe that another back icon is served, despite being at the bottom of the stack.
6. Click the back icon again, and note that this is now replaced with the menu icon which was expected.

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | ^1.5.8
| react-native     | ^0.54.3
| node             | v8.9.4
| npm or yarn      | npm v4.6.1

**Comment by charles-salmon** [comment]
Note that the best workaround I have found for this so far is to replace `props.navigation.state.index === 0 & !props.navigation.state.isTransitioning` with `props.navigation.state.index === 0`. The obvious pitfall that this has is that a "Menu" icon is displayed while a page is being popped off the stack, which is something I'd really like to avoid. I welcome any better ideas!

**Comment by charles-salmon** [comment]
After a lot of perseverance, I managed to work out that this was not in fact a bug, but an implementation issue.

The [API documentation](https://reactnavigation.org/docs/navigation-prop.html#pop) clearly states that `pop()` is a Stack Action, callable from within a page belonging to a `StackNavigator`.

I was wrapping the component which was calling `pop()` using `withNavigation()`, and it therefore did not belong to a `StackNavigator`, so it was unclear which stack needed to be popped off from.

Accordingly, I established that I should have been calling `goBack()` as opposed to `pop()`. D'oh.
